scilpy/tractanalysis/features.py

```diff
@@ -0,0 +1,162 @@
+# -*- coding: utf-8 -*-
+
+from builtins import range
+from itertools import count, takewhile
+import logging
+
+from dipy.segment.clustering import Cluster, QuickBundles
+from dipy.tracking import metrics as tm
+import numpy as np
+
+
+def remove_loops_and_sharp_turns(streamlines,
+                                 max_angle,
+                                 use_qb=False,
+                                 qb_threshold=15.):
+    """
+    Remove loops and sharp turns from a list of streamlines.
+    Parameters
+    ----------
+    streamlines: list of ndarray
+        The list of streamlines from which to remove loops and sharp turns.
+    use_qb: bool
+        Set to True if the additional QuickBundles pass is done.
+        This will help remove sharp turns. Should only be used on
+        bundled streamlines, not on whole-brain tractograms.
+    max_angle: float
+        Maximal winding angle a streamline can have before
+        being classified as a loop.
+    qb_threshold: float
+        Quickbundles distance threshold, only used if use_qb is True.
+    Returns
+    -------
+    A tuple containing
+        list of ndarray: the clean streamlines
+        list of ndarray: the list of removed streamlines, if any
+    """
+
+    loops = []
+    streamlines_clean = []
+    for s in streamlines:
+        if tm.winding(s) >= max_angle:
+            loops.append(s)
+        else:
+            streamlines_clean.append(s)
+
+    # TODO can we use QBx instead?
```

Comment:
If it is not too complicated, can you use QBx ? It could be nice !

Reply:
Just be aware that using QBx will not yield exactly the same results. Thresholds will need to be adjusted.